component/otelcol/connector/grafanacloud/factory.go

```diff
@@ -0,0 +1,35 @@
+package grafanacloud
```

Comment:
IMO it's a bit awkward for a native Flow component to go for the otelcol factory + wrapper approach, which I'd expect to only see if we were wrapping a component made for the pure collector engine. 

If possible, can this be reduced to just a Flow component? That would help cut down the amount of code. For examples,  [otelcol.receiver.prometheus](https://github.com/grafana/agent/blob/main/component/otelcol/receiver/prometheus/prometheus.go) and [otelcol.exporter.prometheus](https://github.com/grafana/agent/blob/main/component/otelcol/exporter/prometheus/prometheus.go) are two native Flow components that deal with OpenTelemetry data.

Reply:
The plan  is to contribute this component upstream to opentelemetry-collector-contrib so otel collector users can use this connector. We didn't want to block the rollout of this component on having it upstreaming first. The plan is to get this in for the next release, port it to contrib, and then import it from contrib at which time we'd be able to remove the connector implementation.

Reply:
OK 👍